plutus-scb-client/src/MainFrame.purs

```diff
@@ -13,28 +13,45 @@ import Control.Monad.Except.Trans (ExceptT, runExceptT)
 import Control.Monad.Reader (class MonadAsk, runReaderT)
 import Control.Monad.State (class MonadState)
 import Control.Monad.State.Extra (zoomStateT)
+import Data.Array as Array
 import Data.Foldable (traverse_)
-import Data.Json.JsonUUID (_JsonUUID)
-import Data.Lens (assign, to, toArrayOf, traversed, view)
+import Data.Lens (assign, modifying, to, toArrayOf, traversed, view)
 import Data.Lens.At (at)
 import Data.Lens.Extra (peruse)
+import Data.Lens.Index (ix)
 import Data.Map as Map
 import Data.Maybe (Maybe(..))
-import Data.UUID as UUID
+import Data.RawJson (RawJson(..))
+import Data.Traversable (for_)
 import Effect.Aff.Class (class MonadAff)
-import Effect.Class (class MonadEffect, liftEffect)
-import Effect.Console (log)
+import Effect.Class (class MonadEffect)
+import Foreign.Generic (encodeJSON)
 import Halogen (Component, hoist)
 import Halogen as H
 import Halogen.HTML (HTML)
+import Language.Plutus.Contract.Effects.ExposeEndpoint (EndpointDescription)
+import Ledger.Ada (Ada(..))
+import Ledger.Extra (adaToValue)
+import Ledger.Value (Value)
 import Network.RemoteData (RemoteData(..), _Success)
 import Network.RemoteData as RemoteData
-import Plutus.SCB.Webserver (SPParams_(SPParams_), getContractByContractinstanceidSchema, getFullreport)
+import Playground.Lenses (_endpointDescription, _getEndpointDescription, _schema)
+import Playground.Types (FunctionSchema, _FunctionSchema)
+import Plutus.SCB.Webserver (SPParams_(..), getContractByContractinstanceidSchema, getFullreport, putContractByContractinstanceidEndpointByEndpointname)
+import Plutus.SCB.Webserver.Types (ContractSignatureResponse(..))
+import Prim.TypeError (class Warn, Text)
+import Schema (FormSchema)
+import Schema.Types (formArgumentToJson, toArgument)
+import Schema.Types as Schema
 import Servant.PureScript.Ajax (AjaxError)
 import Servant.PureScript.Settings (SPSettings_, defaultSettings)
-import Types (HAction(..), Query, State(..), WebData, _annotatedBlockchain, _chainState, _contractInstanceId, _contractSignatures, _csContract, _fullReport, _latestContractStatuses)
+import Types (EndpointForm, HAction(..), Query, State(..), WebData, _annotatedBlockchain, _chainState, _contractInstanceIdString, _contractSignatures, _csContract, _fullReport, _latestContractStatuses)
+import Validation (_arguments)
 import View as View
 
+initialValue :: Value
+initialValue = adaToValue $ Lovelace { getLovelace: 0 }
```

Comment:
I wonder, could some purescript numeric classes be generated if a type has a num class? Then you could have `zero` for example.

Reply:
Yeah, that would work. There's a `Semiring` instance on records too, so it would be:

``` purescript
instance semiringAda :: Semiring Ada where
  zero = wrap zero
  one = wrap one
  add x y = wrap $ add (unwrap x) (unwrap y)
  mul x y = wrap $ mul (unwrap x) (unwrap y)
```

(It can't be derived sadly.)

Do you think this is worth patching into our fork of `purescript-bridge`?

Reply:
Better:

``` purescript
instance semiringAda :: Semiring Ada where
  zero = wrap zero
  one = wrap one
  add = over2 Lovelace add
  mul = over2 Lovelace mul
```

Reply:
yes that's what I mean, I think it would be good in purescript-bridge and in your case it would need to be since orphans, right!

Reply:
doesn't need to be in this PR though